### PR TITLE
mini-css-extract-plugin 

### PR DESCRIPTION
**Pull request by david-fong**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/webpack-contrib/mini-css-extract-plugin#options
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by david-fong** [comment]
Hello! This is my first time contributing to DefinitelyTyped. I wasn't able to run tests for mini-css-extract-plugin because of some errors related to resolving types from `source-map`. Did I miss some step in the environment setup process?

image

File: types/mini-css-extract-plugin/index.d.ts [hunk 1]
```diff
@@ -30,33 +33,75 @@ declare class MiniCssExtractPlugin {
...
+             * Enables/disables ES modules named export for locals.
+             *
+             * Names of locals are converted to camelCase. It is not allowed to use
+             * JavaScript reserved words in CSS class names. Options `esModule` and
+             * `modules.namedExport` in css-loader and MiniCssExtractPlugin.loader
+             * must be enabled.
+             *
+             * @default false
+             */
+            namedExport: boolean;
```
Comment (JounQin) [inline types/mini-css-extract-plugin/index.d.ts:103]: It should be optional.

**Review by JounQin — CHANGES_REQUESTED** [review]

File: types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts [hunk 2]
```diff
@@ -25,8 +25,12 @@ configuration = {
                 test: /\.css$/,
                 use: {
                     loader: MiniCssExtractPlugin.loader,
-                    options: {
+                    options: <MiniCssExtractPlugin.LoaderOptions>{
```
Comment (JounQin) [inline types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:28]: Try to use `as T` instead.

**Review by JounQin — CHANGES_REQUESTED** [review]

**Comment by david-fong** [comment]
@JounQin Thanks for your feedback. I've made the changes you suggested.

I'm still unable to run the tests with `cd types/mini-css-extract-plugin; npm run test mini-css-extract-plugin` due to \@types/webpack importing `source-map`. Do you know if I'm doing something wrong when trying to run the tests? When I cloned the repo, I ran npm install and npm run compile-scripts. Maybe I missed a setup step?

**Comment by david-fong** [comment]
@JounQin @dobogo @skovy @peterblazejewicz can one of you please take a look when you have the time so that this PR doesn't go stale?

File: types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts [hunk 3]
```diff
@@ -27,7 +27,11 @@ configuration = {
                     loader: MiniCssExtractPlugin.loader,
                     options: {
                         publicPath: '/',
-                    },
+                        esModule: true,
+                        modules: {
+                            namedExport: true,
+                        }
+                    } as MiniCssExtractPlugin.LoaderOptions,
```
Comment (peterblazejewicz) [inline types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:34]: you're getting `ERROR: 28:30 no-object-literal-type-assertion` with this one,
either rewrite to use `const ...:LoaderOptions` or add exclusion annotation if acceptable.
THx!

Comment (david-fong) [reply types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts]: Thanks for showing me! I pushed a new commit to address this. Can you please take a look?

File: types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts [hunk 4]
```diff
@@ -3,6 +3,14 @@ import MiniCssExtractPlugin = require('mini-css-extract-plugin');
 
 let configuration: webpack.Configuration;
 
+let loaderOptions: MiniCssExtractPlugin.LoaderOptions = {
```
Comment (peterblazejewicz) [inline types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:6]:
```suggestion
const loaderOptions: MiniCssExtractPlugin.LoaderOptions = {
```

Comment (peterblazejewicz) [reply types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts:6]: You can run 'npm test mini-css-extract-plugin' to preflight your changes locally, btw. thx1

Comment (david-fong) [reply types/mini-css-extract-plugin/mini-css-extract-plugin-tests.ts]: Thanks for making this suggestion! I just realized I forgot to use const like you suggested earlier. Sorry about that.

About local testing, I've been trying to do just that, but I keep getting errors about missing types for the package `source-map`, which is referenced in the webpack typings. Do you know why I am getting those errors? I attached an image of the errors early in this PR thread.

**Review by JounQin — APPROVED** [review]

**Comment by david-fong** [comment]
Ready to merge

**Pull request merged**